geti_sdk/benchmarking/benchmarker.py

```diff
@@ -684,11 +685,335 @@ def run_throughput_benchmark(
                 result_row.update(get_system_info(device=target_device))
                 results.append(result_row)
 
-                # Write results to file
-                if index == 0:
-                    fieldnames = list(result_row.keys())
-                    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
-                    writer.writeheader()
-                writer.writerow(result_row)
-
+        # Write results to file
+        with open(results_file, "w", newline="") as csvfile:
+            fieldnames = list(results[0].keys())
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+            writer.writeheader()
+            writer.writerows(results)
         return results
+
+    def _predict_using_active_model(
+        self,
+        numpy_image: np.ndarray,
+    ) -> dict[str, Any]:
+        """
+        Predict on platform using the active model.
+
+        :param numpy_image: Numpy array containing the image to be predicted on.
+        :return: Dictionary containing the prediction results.
+        """
+        # upload the image
+        image_client = ImageClient(
+            session=self.geti.session,
+            workspace_id=self.geti.workspace_id,
+            project=self.project,
+        )
+        sc_image = image_client.upload_image(numpy_image)
+        # Load the pixel data to visualize the image later on
+        sc_image.get_data(self.geti.session)
+
+        prediction_client = PredictionClient(
+            session=self.geti.session,
+            workspace_id=self.geti.workspace_id,
+            project=self.project,
+        )
+        platform_prediction = prediction_client.get_image_prediction(sc_image)
+        # load active models info
+        active_models = self.model_client.get_all_active_models()
+        result: dict[str, Any] = {}
+        result["prediction"] = platform_prediction
+        result["run_name"] = "online prediction"
+        result["model_1"] = active_models[0].name + " " + active_models[0].precision[0]
+        result["model_1_score"] = active_models[0].performance.score
+        if not self._is_single_task:
+            result["model_2"] = (
+                active_models[1].name + " " + active_models[1].precision[0]
+            )
+            result["model_2_score"] = active_models[1].performance.score
+        return result
+
+    def _pad_image_and_put_caption(
```

Comment:
Could we move this method and the next one to `utils.py`? The `Benchmarker` class is already quite large so I'd like to move code outside as much as possible